samples/caf_sensor_manager/src/modules/sensor_stub_gen.c

```diff
@@ -0,0 +1,172 @@
+		.min  = SENSOR_STUB_SENSOR_FP(SENSOR_STUB_GEN_Z_MIN),
+		.max  = SENSOR_STUB_SENSOR_FP(SENSOR_STUB_GEN_Z_MAX)
+	}
+};
+
+static inline int64_t sensor_val_2_fp(const struct sensor_value *val)
+{
+	return ((int64_t)val->val1 * SENSOR_FRACTIONAL_MAX) + val->val2;
+}
+
+static inline void fp_2_sensor_val(struct sensor_value *out, int64_t val)
+{
+	out->val1 = (int32_t)(val / SENSOR_FRACTIONAL_MAX);
+	out->val2 = (int32_t)(val % SENSOR_FRACTIONAL_MAX);
+}
+
+/* Generate changes on selected channel with given parameters */
+static inline void sensor_stub_gen_chan(struct sensor_stub_gen_ch_data *ch_data,
+					const struct sensor_stub_gen_ch_params *params)
+{
+	int64_t val = sensor_val_2_fp(&ch_data->val);
+
+	if (ch_data->dir) {
+		val += params->step;
+		if (val >= params->max) {
+			ch_data->dir = false;
+		}
+	} else {
+		val -= params->step;
+		if (val <= params->min) {
+			ch_data->dir = true;
+		}
+	}
+
+	fp_2_sensor_val(&ch_data->val, val);
+}
+
+int sensor_stub_gen_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_XYZ:
+	case SENSOR_CHAN_ALL:
+		for (int i = 0; i < SENSOR_STUB_CH_CNT; ++i) {
+			sensor_stub_gen_chan(&gen_data[i], &gen_params[i]);
+		}
+		LOG_DBG("Generated xyz channels: %d.%.6d, %d.%.6d, %d.%.6d",
+			gen_data[SENSOR_STUB_CH_X].val.val1,
+			gen_data[SENSOR_STUB_CH_X].val.val2,
+			gen_data[SENSOR_STUB_CH_Y].val.val1,
+			gen_data[SENSOR_STUB_CH_Y].val.val2,
+			gen_data[SENSOR_STUB_CH_Z].val.val1,
+			gen_data[SENSOR_STUB_CH_Z].val.val2);
+		return 0;
+	case SENSOR_CHAN_ACCEL_X:
+		sensor_stub_gen_chan(&gen_data[SENSOR_STUB_CH_X], &gen_params[SENSOR_STUB_CH_X]);
+		LOG_DBG("Generated x channel: %d.%.6d",
+			gen_data[SENSOR_STUB_CH_X].val.val1,
+			gen_data[SENSOR_STUB_CH_X].val.val2);
+		return 0;
+	case SENSOR_CHAN_ACCEL_Y:
+		sensor_stub_gen_chan(&gen_data[SENSOR_STUB_CH_Y], &gen_params[SENSOR_STUB_CH_Y]);
+		LOG_DBG("Generated y channel: %d.%.6d",
+			gen_data[SENSOR_STUB_CH_Y].val.val1,
+			gen_data[SENSOR_STUB_CH_Y].val.val2);
+		return 0;
+	case SENSOR_CHAN_ACCEL_Z:
+		sensor_stub_gen_chan(&gen_data[SENSOR_STUB_CH_Z], &gen_params[SENSOR_STUB_CH_Z]);
+		LOG_DBG("Generated z channel: %d.%.6d",
+			gen_data[SENSOR_STUB_CH_Z].val.val1,
+			gen_data[SENSOR_STUB_CH_Z].val.val2);
+		return 0;
+	default:
+		return -ENOTSUP;
+	}
+}
+
+int sensor_stub_gen_get(const struct device *dev, enum sensor_channel chan,
+			struct sensor_value *sample)
+{
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_XYZ:
+		for (int i = 0; i < SENSOR_STUB_CH_CNT; ++i) {
+			*sample++ = gen_data[i].val;
+		}
+		return 0;
+	case SENSOR_CHAN_ACCEL_X:
+		*sample = gen_data[SENSOR_STUB_CH_X].val;
+		return 0;
+	case SENSOR_CHAN_ACCEL_Y:
+		*sample = gen_data[SENSOR_STUB_CH_Y].val;
+		return 0;
+	case SENSOR_CHAN_ACCEL_Z:
+		*sample = gen_data[SENSOR_STUB_CH_Z].val;
+		return 0;
+	default:
+		return -ENOTSUP;
+	}
+}
+
+int sensor_stub_gen_init(const struct device *dev)
+{
+	LOG_INF("READY");
```

Comment:
Nothing happens here so maybe `LOG_DBG`?

Reply:
In the sample.yaml there is a check if the sensor_stub is working:
```
regex:
        - "sensor_stub_gen: READY"
        - "main state:READY"
 ```